aws/resource_aws_ebs_volume.go

```diff
@@ -373,3 +371,40 @@ func resourceAwsEbsVolumeDelete(d *schema.ResourceData, meta interface{}) error
 
 	return nil
 }
+
+func resourceAwsEbsVolumeCustomizeDiff(_ context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+	iops := diff.Get("iops").(int)
+	volumeType := diff.Get("type").(string)
+
+	if diff.Id() == "" {
+		// Create.
+
+		// Iops is required for io1 and io2 volumes.
+		// The default for gp3 volumes is 3,000 IOPS.
+		// This parameter is not supported for gp2, st1, sc1, or standard volumes.
+		// Hard validation in place to return an error if IOPs are provided
+		// for an unsupported storage type.
+		// Reference: https://github.com/hashicorp/terraform-provider-aws/issues/12667
+		switch volumeType {
+		case ec2.VolumeTypeIo1, ec2.VolumeTypeIo2:
+			if iops == 0 {
+				return fmt.Errorf("'iops' must be set when 'type' is '%s'", volumeType)
+			}
+
+		case ec2.VolumeTypeGp3:
```

Comment:
If throughput > 1000 it should fail if im not wrong about my assumption here.

Reply:
Also pretty sure you cant go under 125

Reply:
That is validated via `ValidateFunc: validation.IntBetween(125, 1000),`.
The `ValidateFunc` only gets to look at the attribute it's specified on so for cross-attribute validation cases we need to code a `CustomizeDiff` function.
Both run during `terraform plan`.